token/program-2022/src/extension/confidential_transfer/mod.rs

```diff
@@ -6,18 +11,18 @@ use {
     },
     bytemuck::{Pod, Zeroable},
     solana_program::entrypoint::ProgramResult,
-    solana_zk_token_sdk::zk_token_elgamal::pod::{AeCiphertext, ElGamalCiphertext, ElGamalPubkey},
+    solana_zk_token_sdk::zk_token_elgamal::pod::{
+        AeCiphertext as PodAeCiphertext, ElGamalCiphertext, ElGamalPubkey,
+    },
 };
 
 /// Maximum bit length of any deposit or transfer amount
 ///
 /// Any deposit or transfer amount must be less than 2^48
-pub const MAXIMUM_DEPOSIT_TRANSFER_AMOUNT_BIT_LENGTH: usize = 48;
+pub const MAXIMUM_DEPOSIT_TRANSFER_AMOUNT: u64 = (u16::MAX as u64) + (2 << 16) * (u32::MAX as u64);
```

Comment:
If the max is `2^48 - 1`, shouldn't this be?
```suggestion
pub const MAXIMUM_DEPOSIT_TRANSFER_AMOUNT: u64 = (u16::MAX as u64) + (1 << 16) * (u32::MAX as u64);
```

Otherwise you're actually doing `2^49 - 1` -- I tried this out to be sure :sweat_smile:

Reply:
Oh oops! Yes, this actually seems to be a weird 49-bit number with all 1's except the 17th position :sweat_smile: